### PR TITLE
Adds yawn propagation

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
You ever see someone yawn and suddenly want to yawn yourself? Then maybe someone else nearby saw you yawn and yawned too? That's in the game now!

https://streamable.com/uyxbsz

Yawning can make nearby mobs yawn. Also, you can use the classic counter-intelligence method of yawning to check if someone is watching you. If they've examined you in the last 2 seconds and are nearby, they'll be forced to yawn (assuming their cooldown allows it).

The yawn emote now has a 3 second cooldown to perform manually, and you are unable to catch a contagious yawn for 9 seconds after your last one.

## Why It's Good For The Game
Buffs verisimilitude 

## Changelog

:cl: Ryll/Shaps
add: Yawns can now propagate and cause other nearby mobs who see it to yawn as well.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/emote.dm [hunk 1]
```diff
@@ -414,11 +414,60 @@
...
 	emote_type = EMOTE_AUDIBLE
+	cooldown = 3 SECONDS
+
+/datum/emote/living/yawn/run_emote(mob/user, params, type_override, intentional)
+	. = ..()
+	if(!. || !isliving(user))
+		return
+
+	if(!HAS_TRAIT(user, TRAIT_YAWNED_RECENTLY))
+		addtimer(CALLBACK(src, .proc/reset_yawn, user), cooldown * 3)
```
Comment (optimumtact) [inline code/modules/mob/living/emote.dm:435]: isn't there a cooldowns system specifically for this?

**Comment by ATH1909** [comment]
how does this interact with the narcolepsy symptom? will all of the yawns caused by it still go through?

File: code/modules/mob/living/emote.dm [hunk 2]
```diff
@@ -414,11 +414,60 @@
 	key_third_person = "wsmiles"
 	message = "smiles weakly."
 
+/// The base chance for your yawn to propagate to someone else if they're on the same tile as you
+#define YAWN_PROPAGATE_CHANCE_BASE 60
+/// The base chance for your yawn to propagate to someone else if they're on the same tile as you
+#define YAWN_PROPAGATE_CHANCE_DECAY 10
```
Comment (Mothblocks) [inline code/modules/mob/living/emote.dm:420]: undef

File: code/modules/mob/living/emote.dm (hunk 1 above)
Comment (Ryll-Ryll) [reply code/modules/mob/living/emote.dm:435]: Oh yeah that's right duh

Comment (Ryll-Ryll) [reply code/modules/mob/living/emote.dm:435]: Wait no, cause emote datums are singleton, and it seems silly to add a whole new var to /mob/living just to track cooldowns for propagated yawns.

**Comment by Coastfront** [comment]
I have never felt this in my entire life, what the fuck?

**Comment by ike709** [comment]
> I have never felt this in my entire life, what the fuck? nor have i ever heard of this.

https://pubmed.ncbi.nlm.nih.gov/33897548/

**Pull request merged**